packages/nodes-base/nodes/ItemLists/V3/helpers/utils.ts

```diff
@@ -57,3 +64,26 @@ export const prepareFieldsArray = (fields: string | string[], fieldName = 'Field
 		`The \'${fieldName}\' parameter must be a string of fields separated by commas or an array of strings.`,
 	);
 };
+
+const returnRegExp = /\breturn\b/g;
+
+export function sortByCode(
+	this: IExecuteFunctions,
+	items: INodeExecutionData[],
+): INodeExecutionData[] {
+	const code = this.getNodeParameter('code', 0) as string;
+	if (!returnRegExp.test(code)) {
+		throw new NodeOperationError(
+			this.getNode(),
+			"Sort code doesn't return. Please add a 'return' statement to your code",
+		);
+	}
+
+	const mode = this.getMode();
+	const vm = new NodeVM({
+		console: mode === 'manual' ? 'redirect' : 'inherit',
+		sandbox: { items },
+	});
+
+	return vm.run(`module.exports = items.sort((a, b) => { ${code} })`);
```

Comment:
is it deliberately not async anymore?

Reply:
yeah. I don't see why anyone would be using promises in sorting code. but I should also not make this a breaking changes. will revert.

Reply:
nevermind. I just remembered. in the old code `async` code doesn't work, as the `newItems.sort( (a,b) => { ${code} })` block creates it's own non-async arrow function, which prevents the user from using `await` in their code.